MetaMask/metamask-extension
# feat(1463): introduce dapp viewed metrics events when connected to a dapp

## **Description**
Part 1 of https://app.zenhub.com/workspaces/metamask-extension-platform-655e3b56511c30108d50e7b8/issues/gh/metamask/metamask-planning/1463

We want to add `Dapp Viewed` event for several actions. The first one introduced in this PR is emit metrics when connecting to a dapp successfully. Properties are:
```
1. is_first_visit : always true
2. number_of_accounts - number of accounts in wallet
3. number_of_accounts_connected - number of connected accounts to Dapp
```



## **Related issues**

Fixes: https://app.zenhub.com/workspaces/metamask-extension-platform-655e3b56511c30108d50e7b8/issues/gh/metamask/metamask-planning/1463

## **Manual testing steps**

1. Onboard MM
2. Opt in metrics `Participate in MetaMetrics`
3. Open a dapp (eg: [uniswap](https://app.uniswap.org/swap) or [compound)](https://app.compound.finance/?market=usdc-mainnet) 
4. Track events in network with keyword `batch`
5. You would only see `Dapp_viewed` once after success connection, comes with properties as 
- is_first_visit: true (always)
- number_of_accounts
- number_of_accounts_connected

## **Screenshots/Recordings**



### **Before**



### **After**



https://github.com/MetaMask/metamask-extension/assets/12678455/5db37d96-65b9-405b-9923-91e1e3c10832


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
